src/main/kotlin/agallipo/BoardingCard.kt

```diff
@@ -1,59 +1,59 @@
-package agallipo
+    package agallipo
 
-import java.io.File
-import java.io.InputStream
+    import java.io.File
+    import java.io.InputStream
+    //created by alex :)
+    data class BoardingCard(var origin: String?, var destination: String?,  var transport: String?, var gate: String?, var seat: String?)
+    //data class CardInformation(val flight: String?, val gate: String?, val seat: String?)
 
-data class BoardingCard(var origin: String?, var destination: String?,  var transport: String?, var gate: String?, var seat: String?)
-//data class CardInformation(val flight: String?, val gate: String?, val seat: String?)
 
-
-interface BoardingCardsReader {
-    fun readCards() : MutableList<String>
-}
-
-class BoardingCardsFile(val pathname: String) : BoardingCardsReader {
-    override fun readCards(): MutableList<String> {
-        val inputStream: InputStream = File(pathname).inputStream()
-        val lines = mutableListOf<String>()
-        inputStream.bufferedReader().forEachLine(lines::add)
-        return lines
+    interface BoardingCardsReader {
+        fun readCards() : MutableList<String>
     }
-}
-
-interface BoardingCardClassifier {
-    fun parseLine(boardingCardsFile: BoardingCardsFile) : MutableList<BoardingCard?> {
-        var boardingCards = mutableListOf<BoardingCard?>()
-        var boardingCard: BoardingCard? = BoardingCard(null,null,null,null,null)
-        var lines = boardingCardsFile.readCards()
-
-        for (line in lines ) {
-            if (line.compareTo("{") != 0 && line.compareTo("}") != 0)
-               setCardValues(line, boardingCard)
-            if (line.compareTo("}") == 0)
-                boardingCards.add(boardingCard!!.copy())
+
+    class BoardingCardsFile(val pathname: String) : BoardingCardsReader {
+        override fun readCards(): MutableList<String> {
+            val inputStream: InputStream = File(pathname).inputStream()
+            val lines = mutableListOf<String>()
+            inputStream.bufferedReader().forEachLine(lines::add)
+            return lines
         }
-        return boardingCards
     }
 
-    fun setCardValues(line: String, boardingCard: BoardingCard?){
+    interface BoardingCardClassifier {
+        fun parseLine(boardingCardsFile: BoardingCardsFile) : MutableList<BoardingCard?> {
+            var boardingCards = mutableListOf<BoardingCard?>()
+            var boardingCard: BoardingCard? = BoardingCard(null,null,null,null,null)
```

Comment:
I would create the object inside the loop, just to ensure you have a clean object each iteration